### PR TITLE
MAINT: speed up comb(..., exact=True) calculation for small N, k

### PR DESCRIPTION
**Pull request by ev-br**
An additional factor for small N and/or k where all intermediaries can fit to a Py_ssize_t.

### CODE REVIEW

File: scipy/special/_comb.pyx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+from libc.math cimport log
+cdef extern from "limits.h":
+    Py_ssize_t LONG_MAX
+
+
+# Compute N choose k in integer arithmetic.
+def _comb_int(Py_ssize_t N, Py_ssize_t k):
```
Comment (pv) [inline scipy/special/_comb.pyx]: suggest dropping Py_ssize_t declaration, the old code works for any N, k

File: scipy/special/_comb.pyx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        return 0
+
+    M = N + 1
+    nterms = min(k, N - k)  
+
+    # delegate to fully typed version unless overflow
+    # The condition is rather loose:
+    #    numerator < M**nterms
+    # hence log(numerator) < nterms * log(M) < log(LONG_MAX)
+    if nterms*log(1.0*M) < log(1.0*LONG_MAX):
```
Comment (pv) [inline scipy/special/_comb.pyx]: Better: `M < pow(LONG_MAX, 1.0/nterms)` --- works also if M is Python long int.
Handle also nterms == 0.

File: scipy/special/_comb.pyx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+from libc.math cimport log
+cdef extern from "limits.h":
+    Py_ssize_t LONG_MAX
+
+
+# Compute N choose k in integer arithmetic.
+def _comb_int(Py_ssize_t N, Py_ssize_t k):
+    cdef:
+        object numerator, denominator
+        Py_ssize_t j, M, nterms
```
Comment (pv) [inline scipy/special/_comb.pyx]: j, M, nterms should also be object.

File: scipy/special/_comb.pyx (hunk 2 above)
Comment (pv) [inline scipy/special/_comb.pyx]: Also, Py_ssize_t can in principle be smaller than long.
So either the "unsafe" routine should be using long, or the LONG_MAX constant be replaced with the size constant corresponding to Py_ssize_t. I'd suggest the former.

**Comment by ev-br** [comment]
OK, updated.

**Comment by pv** [comment]
thanks. optimized a bit further in ab1daa2

**Pull request closed without merging**